dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java

```diff
@@ -167,6 +264,18 @@ public EPersonRest findByEmail(@Parameter(value = "email", required = true) Stri
     @PreAuthorize("hasPermission(#uuid, 'EPERSON', #patch)")
     protected void patch(Context context, HttpServletRequest request, String apiCategory, String model, UUID uuid,
                          Patch patch) throws AuthorizeException, SQLException {
+        if (StringUtils.isNotBlank(request.getParameter("token"))) {
+            boolean passwordChangeFound = false;
+            for (Operation operation : patch.getOperations()) {
+                if (StringUtils.equalsIgnoreCase(operation.getPath(), "/password")) {
+                    passwordChangeFound = true;
+                }
+            }
+            if (!passwordChangeFound) {
+                throw new AccessDeniedException("Couldn't perform the patch as a token with provided without " +
```

Comment:
I think this error message needs improving.  But, I'm not sure how to reword it as I'm not sure I understand what is going on here.  Is this code trying to check if this is a self-registration & verify then that you *must* change your password?  Or am I misunderstanding why a Token requires a password change?

Reply:
This has been adjusted